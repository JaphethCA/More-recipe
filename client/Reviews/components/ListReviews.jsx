import React from 'react';
import PropTypes from 'prop-types';

import Review from './Review';
import '../reviews.scss';


const propTypes = {
  reviews: PropTypes.arrayOf(PropTypes.any).isRequired
};

/**
 * @description displays list of reviews
 * @returns {ReactElement} markup
 */
const ListReviews = ({ reviews }) => (
  <div className="review-list">
    <h4 >Reviews</h4>
    {reviews.length < 1 && <span className="no-reviews">No reviews yet!</span>}
    <div className="row">
      {reviews.map(review => (
        <div key={review.id} className="col s12 m12 l6">
          <Review review={review} />
        </div>
      ))
      }
    </div>
  </div>
);

ListReviews.propTypes = propTypes;

export default ListReviews;
