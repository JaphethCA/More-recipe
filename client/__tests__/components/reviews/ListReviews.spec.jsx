import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import ListReviews from '../../../Reviews/components/ListReviews';
import mockData from '../../__mock__/mockData';


const props = {
  reviews: [mockData.Review]
};

describe('<ListReviews /> ', () => {
  it('should render without exploding', () => {
    const wrapper = shallow(<ListReviews {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('h4').length).toBe(1);
    expect(wrapper.find('.review-list').length).toBe(1);
    expect(wrapper.find('Review').length).toBe(1);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<ListReviews {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

