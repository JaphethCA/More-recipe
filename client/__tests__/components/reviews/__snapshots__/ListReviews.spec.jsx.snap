// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListReviews />  should match match snapshot 1`] = `
<div
  className="review-list"
>
  <h4>
    Reviews
  </h4>
  <div
    className="row"
  >
    <div
      className="col s12 m12 l6"
    >
      <Review
        review={
          Object {
            "User": "chidex",
            "content": "this is a great recipe",
            "createdAt": "2017-12-30T22:05:17.204Z",
          }
        }
      />
    </div>
  </div>
</div>
`;
